lib/Parse/ParseExpr.cpp

```diff
@@ -3290,12 +3293,13 @@ ParserStatus Parser::parseExprList(tok leftTok, tok rightTok,
   StructureMarkerRAII ParsingExprList(*this, Tok);
   
   leftLoc = consumeToken(leftTok);
-  return parseList(rightTok, leftLoc, rightLoc, /*AllowSepAfterLast=*/false,
+  return parseList(rightTok, leftLoc, rightLoc, /*AllowSepAfterLast=*/true,
```

Comment:
I don't know the `yield` statement but if it's something like `yield(...)` I believe it should. My plan is to go back to the swift-syntax PR to fix any inconsistent once this PR is approved.

Reply:
`yield` can be followed by an expression list without parentheses. Eg. `yield x, y` is valid. But `yield x, ` should be disallowed.

Reply:
The accepted portions of the proposal are exclusively symmetrically delimited comma-separated lists (i.e., those surrounded by `(...)`, `[...]`, `<...>`, etc.), so this would need to be gated behind a flag.

Reply:
@ahoppen do you have an example for this? I tried the following but it doesn't parse::
```
struct Foo {
  var x: Int
  var y: Int {
    _read { yield x, } // ❌ consecutive statements on a line must be separated by ';' | expected expression
  }
}
```

Reply:
Ah, sorry. I did misremember how `yield` works. It does require parentheses if multiple expressions should be yielded, so this is not an issue. Apologies for the confusion.